Client/Frontend/Browser/BackForwardListView.swift

```diff
@@ -0,0 +1,196 @@
+        super.init(frame: CGRect.zero)
+        self.backgroundColor = UIColor.clearColor()
+        
+        self.profile = profile
+        self.isPrivate = isPrivate
+        
+        tableView = UITableView()
+        tableView.separatorStyle = .None
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.tableFooterView = UIView(frame: CGRectMake(0, 0, tableView.frame.size.width, 1))
+        tableView.alwaysBounceVertical = false
+        
+        if isPrivate {
+            tableView.backgroundColor = UIColor.init(colorLiteralRed: 0.5, green: 0.5, blue: 0.5, alpha: 0.4)
+            let blurEffect = UIBlurEffect(style: .Dark)
+            let blurEffectView = UIVisualEffectView(effect: blurEffect)
+            tableView.backgroundView = blurEffectView
+        }
+        else {
+            tableView.backgroundColor = UIColor.init(colorLiteralRed: 1, green: 1, blue: 1, alpha: 0.4)
+            let blurEffect = UIBlurEffect(style: .ExtraLight)
+            let blurEffectView = UIVisualEffectView(effect: blurEffect)
+            tableView.backgroundView = blurEffectView
+        }
+        
+        let sql = profile.favicons as! SQLiteHistory
+        var urls: [String] = [String]()
+        
+        for page in backForwardList.forwardList.reverse() {
+            urls.append(page.URL.absoluteString)
+            listData.append((page, .Forward))
+        }
+        
+        var currentRow = 0
+        if let currentPage = backForwardList.currentItem {
+            currentRow = listData.count
+            urls.append(currentPage.URL.absoluteString)
+            listData.append((currentPage, .Current))
+        }
+        for page in backForwardList.backList.reverse() {
+            urls.append(page.URL.absoluteString)
+            listData.append((page, .Backward))
+        }
+        
+        listData = listData.filter { !(($0.item.title ?? "").isEmpty)}
+        
+        let deferred = sql.getSitesForURLs(urls)
+        
+        deferred.uponQueue(dispatch_get_main_queue()) { result in
+            if let cursor = result.successValue {
+                for cursorSite in cursor {
+                    if let site = cursorSite, let url = site?.url {
+                        self.sites[url] = site
+                    }
+                }
+                self.tableView.reloadData()
+            }
+        }
+        
+        addSubview(tableView)
+        tableView.snp_makeConstraints { make in
+            make.height.equalTo(0)
+            make.left.right.bottom.equalTo(self)
+        }
+        
+        if currentRow > 1 {
+            self.tableView.reloadData()
+            let moveToIndexPath = NSIndexPath(forRow: currentRow-2, inSection: 0)
+            self.tableView.reloadRowsAtIndexPaths([moveToIndexPath], withRowAnimation: .None)
+            self.tableView.scrollToRowAtIndexPath(moveToIndexPath, atScrollPosition: UITableViewScrollPosition.Middle, animated: false)
+        }
```

Comment:
Nit: In general this init method does a lot of things. Might be easier to break it apart into small, single purpose functions to bring some clarity.